### PR TITLE
Try with contracted graph

### LINKED ISSUE #14: Benchmark dodgr with contracted graph

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

File: dodgr.Rmd [hunk 1]
```diff
@@ -93,6 +95,19 @@ Plot
 plot(paths2)
 ```
 
+# On contracted graph
+
+```{r}
+o <- match_pts_to_verts(dodgr_vertices(graph), origins, connected = TRUE)
```
Comment (mpadge) [inline dodgr.Rmd]:
```suggestion
v <- dodgr_vertices(graph)
o <- v$id[match_pts_to_verts(v, origins, connected = TRUE)]
```

File: dodgr.Rmd [hunk 2]
```diff
@@ -93,6 +95,19 @@ Plot
...
+d <- match_pts_to_verts(dodgr_vertices(graph), destinations, connected = TRUE)
```
Comment (mpadge) [inline dodgr.Rmd]:
```suggestion
d <- v$id[match_pts_to_verts(v, destinations, connected = TRUE)]
```
Sorry, that was my bad - those `o` and `d` values should be vertex IDs. I've also updated code to issue message when trying to do the former way.

Comment (Robinlovelace) [reply dodgr.Rmd]: Ace, thanks! Will try now. Error could be a bit more informative, right?

**Comment by Robinlovelace** [comment]
Could you reproduce the whole thing with RMarkdown @mpadge? I got...

```
17/20                  
18/20 [unnamed-chunk-6]


 *** caught segfault ***
address 0x68b784a85048, cause 'memory not mapped'


Traceback:
 1: rcpp_get_paths_pairwise(graph, vert_map, from_index$index, to_index$index,     heap)
 2: dodgr_paths(graphc, o, d, pairwise = TRUE)
 3: system.time({    paths <- dodgr_paths(graphc, o, d, pairwise = TRUE)})
 4: eval(expr, envir, enclos)
 5: eval(expr, envir, enclos)
 6: eval_with_user_handlers(expr, envir, enclos, user_handlers)
 7: withVisible(eval_with_user_handlers(expr, envir, enclos, user_handlers))
 8: withCallingHandlers(withVisible(eval_with_user_handlers(expr,     envir, enclos, user_handlers)), warning = wHandler, error = eHandler,     message = mHandler)
 9: handle(ev <- withCallingHandlers(withVisible(eval_with_user_handlers(expr,     envir, enclos, user_handlers)), warning = wHandler, error = eHandler,     message = mHandler))
10: timing_fn(handle(ev <- withCallingHandlers(withVisible(eval_with_user_handlers(expr,     envir, enclos, user_handlers)), warning = wHandler, error = eHandler,     message = mHandler)))
11: evaluate_call(expr, parsed$src[[i]], envir = envir, enclos = enclos,     debug = debug, last = i == length(out), use_try = stop_on_error !=         2L, keep_warning = keep_warning, keep_message = keep_message,     log_echo = log_echo, log_warning = log_warning, output_handler = output_handler,     include_timing = include_timing)
12: evaluate::evaluate(...)
13: evaluate(code, envir = env, new_device = FALSE, keep_warning = if (is.numeric(options$warning)) TRUE else options$warning,     keep_message = if (is.numeric(options$message)) TRUE else options$message,     stop_on_error = if (is.numeric(options$error)) options$error else {        if (options$error && options$include)             0L        else 2L    }, output_handler = knit_handlers(options$render, options))
14: in_dir(input_dir(), expr)
15: in_input_dir(evaluate(code, envir = env, new_device = FALSE,     keep_warning = if (is.numeric(options$warning)) TRUE else options$warning,     keep_message = if (is.numeric(options$message)) TRUE else options$message,     stop_on_error = if (is.numeric(options$error)) options$error else {        if (options$error && options$include)             0L        else 2L    }, output_handler = knit_handlers(options$render, options)))
16: eng_r(options)
17: block_exec(params)
18: call_block(x)
19: process_group(group)
20: withCallingHandlers(if (tangle) process_tangle(group) else process_group(group),     error = function(e) if (xfun::pkg_available("rlang", "1.0.0")) rlang::entrace(e))
21: xfun:::handle_error(withCallingHandlers(if (tangle) process_tangle(group) else process_group(group),     error = function(e) if (xfun::pkg_available("rlang", "1.0.0")) rlang::entrace(e)),     function(loc) {        setwd(wd)        write_utf8(res, output %n% stdout())        paste0("\nQuitting from lines ", loc)    }, if (labels[i] != "") sprintf(" [%s]", labels[i]), get_loc)
22: process_file(text, output)
23: knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
24: rmarkdown::render("/home/robin/github/itsleeds/routingday/dodgr.Rmd")
25: eval(parse(text = .vsc.knit_command))
26: eval(parse(text = .vsc.knit_command))
27: cat(.vsc.knit_lim, eval(parse(text = .vsc.knit_command)), .vsc.knit_lim,     sep = "", file = stdout())
An irrecoverable exception occurred. R is aborting now ...

[VSC-R] dodgr.Rmd process exited from signal 'SIGSEGV'
[VSC-R] terminating R process
```

**Pull request merged**